pkgs/stdenv/generic/setup.sh

```diff
@@ -334,6 +334,35 @@ appendToVar() {
     fi
 }
 
+# Add flag to a given variable if the variable does not yet contain the flag.
+# The variable to add to can be either an array or a string.
+# Usage: addFlagTo NIX_CFLAGS_COMPILE "-Wblabla"
+#        addFlagTo NIX_CFLAGS_COMPILE "-isystem foobar"
+#        addFlagTo NIX_CFLAGS_COMPILE "--foo=\"bar baz\""
+addFlagTo() {
+    local varname="$1"
+    # Use a different name as in appendToVar to avoid a name clash
+    local -n _nameref="$1"
+
+        # check if variable already exist and if it does then do extra checks
+    if declare -p "$1" 2> /dev/null | grep -q '^'; then
+        type="$(declare -p "$1")"
+        if [[ "$type" =~ "declare -A" ]]; then
+            echo "addFlagTo(): ERROR: trying to use addFlagTo on an associative array." >&2
+            return 1
+        fi
+    fi
+
+    shift
+
+    # If the variable already contains the flag, we don't add it again
+    if [[ "${_nameref[*]}" =~ ^(.* )*"${*}"( .*)*$ ]]; then
```

Comment:
```suggestion
    if [[ "${_nameref[*]}" =~ (^| )"$2"( |$) ]]; then
```

(Assuming that only one argument is supported and consequently the `shift` above is removed.)